app/models/settings/smtp.rb

```diff
@@ -14,14 +14,24 @@ class SMTP < Base
 
     class << self
       def settings
-        return sendgrid_settings if ENV["SENDGRID_API_KEY"].present?
+        if provided_minimum_settings?
+          custom_provider_settings
+        else
+          fallback_sendgrid_settings
+        end
+      end
 
-        keys.index_with { |k| public_send(k) }.symbolize_keys
+      def provided_minimum_settings?
+        address.present? && user_name.present? && password.present?
       end
 
       private
 
-      def sendgrid_settings
+      def custom_provider_settings
+        to_h
+      end
```

Comment:
This feels a bit overdone but I like the explicit description of the method.

Reply:
It is more maintable. If we ever add new settings on the model that don't need to be included we can simply update this to `to_h.except(...)` and not change any call sites.